### PR TITLE
gruvbox-plus-icons: init at unstable-2023-11-30

### PR DESCRIPTION
**Pull request by RGBCube**
## Description of changes

Added gruvbox-icons-plus and material-cursors

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -29389,6 +29389,8 @@ with pkgs;
     inherit (plasma5Packages) breeze-icons;
   };
 
+  gruvbox-plus-icons = callPackage ../data/icons/gruvbox-plus-icons { };
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:29392]: move to `pkgs/by-name` and remove the changes in this file

File: pkgs/data/icons/material-cursors/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Material cursors for Linux";
+    homepage = "https://github.com/varlesh/material-cursors";
+    license = licenses.gpl3Only;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:42]: keep this on one line, and use `lib.maintainers`

File: pkgs/data/icons/material-cursors/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  version = "2023-11-30";
+
+  src = fetchFromGitHub {
+    owner = "varlesh";
+    repo = "material-cursors";
+    rev = "2a5f302fefe04678c421473bed636b4d87774b4a";
+    hash = "sha256-uC2qx3jF4d2tGLPnXEpogm0vyC053MvDVVdVXX8AZ60=";
+  };
+
+  nativeBuildInputs = with pkgs; [
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:14]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/data/icons/material-cursors/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    inkscape
+    xorg.xcursorgen
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:16]:
```suggestion
    xcursorgen
```

File: pkgs/data/icons/material-cursors/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{ lib, pkgs, stdenvNoCC, fetchFromGitHub }:
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:1]:
```suggestion
{ lib, inkscape, xcursorgen, stdenvNoCC, fetchFromGitHub }:
```

File: pkgs/data/icons/material-cursors/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+stdenvNoCC.mkDerivation {
+  pname   = "material-cursors";
+  version = "2023-11-30";
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:5]:
```suggestion
  version = "unstable-2023-11-30";
```

File: pkgs/data/icons/gruvbox-plus-icons/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
+
+stdenvNoCC.mkDerivation {
+  pname = "gruvbox-plus-icons";
+  version = "2023-11-30";
```
Comment (eclairevoyant) [inline pkgs/data/icons/gruvbox-plus-icons/default.nix:5]:
```suggestion
  version = "unstable-2023-11-30";
```

File: pkgs/data/icons/gruvbox-plus-icons/default.nix [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Gruvbox Plus icon pack for Linux desktops based on the Gruvbox color theme";
+    homepage = "https://github.com/SylEleuth/gruvbox-plus-icon-pack";
+    license = licenses.gpl3Only;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
```
Comment (eclairevoyant) [inline pkgs/data/icons/gruvbox-plus-icons/default.nix:28]: see above

File: pkgs/data/icons/material-cursors/default.nix [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/icons
+    cp -r dist/* $out/share/icons/
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:37]:
```suggestion
  meta = {
```

File: pkgs/data/icons/material-cursors/default.nix [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    description = "Material cursors for Linux";
+    homepage = "https://github.com/varlesh/material-cursors";
+    license = licenses.gpl3Only;
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:40]:
```suggestion
    license = lib.licenses.gpl3Only;
```

File: pkgs/data/icons/material-cursors/default.nix [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    platforms = platforms.linux;
```
Comment (eclairevoyant) [inline pkgs/data/icons/material-cursors/default.nix:41]:
```suggestion
    platforms = lib.platforms.linux;
```

File: pkgs/data/icons/gruvbox-plus-icons/default.nix [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/icons
+    cp -r Gruvbox-Plus-Dark $out/share/icons/
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/data/icons/gruvbox-plus-icons/default.nix:23]: see above

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! Some feedback below

File: pkgs/by-name/gr/gruvbox-plus-icons/package.nix [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    runHook preInstall
+
+    mkdir -p $out/share/icons
+    cp -r Gruvbox-Plus-Dark $out/share/icons/
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Gruvbox Plus icon pack for Linux desktops based on the Gruvbox color theme";
```
Comment (eclairevoyant) [inline pkgs/by-name/gr/gruvbox-plus-icons/package.nix:24]:
```suggestion
    description = "Icon pack for Linux desktops based on the Gruvbox color theme";
```

I missed this earlier, desc shouldn't have the package name

File: pkgs/by-name/gr/gruvbox-plus-icons/package.nix [hunk 14]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
+
+stdenvNoCC.mkDerivation {
+  pname = "gruvbox-plus-icons";
+  version = "2023-11-30";
```
Comment (eclairevoyant) [inline pkgs/by-name/gr/gruvbox-plus-icons/package.nix:5]:
```suggestion
  version = "unstable-2023-11-30";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Comment by RGBCube** [comment]
Ok to merge now?

**Review by Gerg-L — APPROVED** [review]
looks good to me

**Review by emilytrau — APPROVED** [review]

**Comment by emilytrau** [comment]
Thanks and welcome to nixpkgs!

**Pull request merged**